### PR TITLE
[REVIEW] Add support for percentile dispatch in `dask_cudf`

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR adds support for percentile dispatch in `dask_cudf`. 

Upstream changes needed: https://github.com/dask/dask/pull/8029

### CODE REVIEW

File: python/dask_cudf/dask_cudf/backends.py [hunk 1]
```diff
@@ -256,6 +258,51 @@ def is_categorical_dtype_cudf(obj):
     return cudf.utils.dtypes.is_categorical_dtype(obj)
 
 
+try:
+    from dask.dataframe.dispatch import percentile_dispatch
+
+    @percentile_dispatch.register((cudf.Series, cp.ndarray, cudf.Index))
+    def percentile_cudf(a, q, interpolation="linear"):
+        a = cudf.Series(a)
```
Comment (marlenezw) [inline python/dask_cudf/dask_cudf/backends.py:266]: What are the inputs `a` and `q`? This looks good, but maybe we can add some clearer documentation on this?

**Review by marlenezw — APPROVED** [review]
Overall looks good to me, just a minor suggestion.

File: python/dask_cudf/dask_cudf/backends.py [hunk 2]
```diff
@@ -256,6 +258,51 @@ def is_categorical_dtype_cudf(obj):
...
+        # a is series.
+        n = len(a)
+        if not len(a):
+            return None, n
+        if isinstance(q, Iterator):
+            q = list(q)
+
+        if cudf.utils.dtypes.is_categorical_dtype(a.dtype):
+            result = cp.percentile(a.cat.codes, q, interpolation=interpolation)
+            import pandas as pd
```
Comment (marlenezw) [inline python/dask_cudf/dask_cudf/backends.py:276]: Also, should we be importing pandas here again? Looks like it's already imported above.

**Comment by quasiben** [comment]
rerun tests

File: python/dask_cudf/dask_cudf/backends.py (hunk 2 above)
Comment (galipremsagar) [reply python/dask_cudf/dask_cudf/backends.py]: Removed it.

File: python/dask_cudf/dask_cudf/backends.py (hunk 1 above)
Comment (galipremsagar) [reply python/dask_cudf/dask_cudf/backends.py:268]: Added a comment pointing to the same.

**Review by quasiben — APPROVED** [review]
Thanks @galipremsagar!

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**